python-package/lightgbm/sklearn.py

```diff
@@ -401,81 +442,128 @@ def get_meta_data(collection, i):
         self._Booster = train(params, train_set,
                               self.n_estimators, valid_sets=valid_sets, valid_names=eval_names,
                               early_stopping_rounds=early_stopping_rounds,
-                              evals_result=evals_result, fobj=self.fobj, feval=feval,
+                              evals_result=evals_result, fobj=self._fobj, feval=feval,
                               verbose_eval=verbose, feature_name=feature_name,
                               categorical_feature=categorical_feature,
                               callbacks=callbacks)
 
         if evals_result:
-            self.evals_result = evals_result
+            self._evals_result = evals_result
 
         if early_stopping_rounds is not None:
-            self.best_iteration = self._Booster.best_iteration
-        self.best_score = self._Booster.best_score
+            self._best_iteration = self._Booster.best_iteration
+        self._best_score = self._Booster.best_score
```

Comment:
Should the L455 be into the `if` statement?

Reply:
@StrikerRUS yes.